tests_integration/nativeapp/test_event_sharing.py

```diff
@@ -0,0 +1,253 @@
+import pytest
+import yaml
+
+from tests.nativeapp.factories import (
+    ApplicationEntityModelFactory,
+    ApplicationPackageEntityModelFactory,
+    ProjectV2Factory,
+)
+from tests_integration.test_utils import row_from_snowflake_session
+
+
+def build_manifest(configuration: dict):
+    manifest_json = {
+        "manifest_version": "1",
+        "artifacts": {
+            "setup_script": "setup.sql",
+            "readme": "README.md",
+        },
+        "configuration": configuration,
+    }
+
+    return yaml.dump(manifest_json)
+
+
+@pytest.fixture
+def events_assertion(snowflake_session, resource_suffix):
+    def _events_assertion(events):
+        events_in_app = row_from_snowflake_session(
+            snowflake_session.execute_string(
+                f"show telemetry event definitions in application myapp{resource_suffix}",
+            )
+        )
+
+        assert sorted(events_in_app, key=lambda x: x["type"]) == sorted(
+            events,
+            key=lambda x: x["type"],
+        )
+
+    return _events_assertion
+
+
+def assert_events_in_app(snowflake_session, resource_suffix, events):
+    events_in_app = row_from_snowflake_session(
+        snowflake_session.execute_string(
+            f"show telemetry event definitions in application myapp{resource_suffix}",
+        )
+    )
+
+    assert sorted(events_in_app, key=lambda x: x["type"]) == sorted(
+        events,
+        key=lambda x: x["type"],
+    )
+
+
+@pytest.mark.integration
+def test_given_event_sharing_with_mandatory_events_and_sharing_allowed_then_success(
+    temp_dir, runner, events_assertion
+):
+    manifest_yml = build_manifest(
+        {
+            "telemetry_event_definitions": [
+                {"type": "ERRORS_AND_WARNINGS", "sharing": "MANDATORY"},
+                {"type": "DEBUG_LOGS", "sharing": "OPTIONAL"},
+            ]
+        }
+    )
+    ProjectV2Factory(
+        pdf__entities=dict(
+            pkg=ApplicationPackageEntityModelFactory(
+                identifier="my_pkg",
+                artifacts=[{"src": "*", "dest": "./"}],
+            ),
+            app=ApplicationEntityModelFactory(
+                identifier="myapp",
+                fromm__target="pkg",
+                telemetry={
+                    "share_mandatory_events": True,
+                    "optional_shared_events": ["DEBUG_LOGS"],
+                },
+            ),
+        ),
+        files={
+            "setup.sql": "CREATE OR ALTER VERSIONED SCHEMA core;",
+            "README.md": "\n",
+            "manifest.yml": manifest_yml,
+        },
+    )
+
+    result = runner.invoke_with_connection(["app", "run"])
```

Comment:
don't we need to tear down these apps? I thought we had a context managed or a fixture for that, but I don't see them used here.